src/ngx_http_lua_ssl_certby.c

```diff
@@ -1294,4 +1294,119 @@ ngx_http_lua_ffi_set_priv_key(ngx_http_request_t *r,
 }
 
 
+static int
+ngx_ssl_verify_callback(int ok, X509_STORE_CTX *x509_store)
+{
+    /*
+     * we never terminate handshake here and user can later use
+     * $ssl_client_verify to check verification result.
+     *
+     * this is consistent with Nginx behavior.
+     */
+    return 1;
+}
+
+
+int
+ngx_http_lua_ffi_ssl_verify_client(ngx_http_request_t *r, void *ca_certs,
+    int depth, char **err)
+{
+    ngx_ssl_conn_t              *ssl_conn;
+    ngx_http_ssl_srv_conf_t     *sscf;
+    STACK_OF(X509)              *chain = ca_certs;
+    STACK_OF(X509_NAME)         *name_chain = NULL;
+    X509                        *x509 = NULL;
+    X509_NAME                   *subject = NULL;
+    X509_STORE                  *ca_store = NULL;
+#ifdef OPENSSL_IS_BORINGSSL
+    size_t                      i;
+#else
+    int                         i;
+#endif
+
+    if (r->connection == NULL || r->connection->ssl == NULL) {
+        *err = "bad request";
+        return NGX_ERROR;
+    }
+
```

Comment:
As [the document describe](https://github.com/openresty/lua-resty-core/pull/289/files#diff-a69ae86a1de8bc9594d4d4f6dca89709R482), here should check the phase like [ngx.exit](https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_control.c#L363). And now the function can be run other phase such as `content_by_lua_block`.

Reply:
I see other SSL APIs don't have this check either. Is this simply because of some historical reasons?

Reply:
Check added.

Reply:
> I see other SSL APIs don't have this check either. Is this simply because of some historical reasons?

Yes, I agree, but I think we should fix these inconsistencies(such as [set_der_cert](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#set_der_cert)) with the documentation and add some test cases, right? @doujiang24